### PR TITLE
Use the right mask for case level

### LINKED ISSUE #2884: Collator's `CaseLevel` option not working
The following snippet panics:

```Rust
use icu_collator::{CaseLevel, Collator, CollatorOptions, Strength};
use icu_locid::locale;

fn main() {
    let data = icu_testdata::unstable();
    let collator = Collator::try_new_unstable(&data, &locale!("es").into(), {
        let mut opts = CollatorOptions::new();
        opts.strength = Some(Strength::Primary);
        opts.case_level = Some(CaseLevel::On);
        opts
    }).unwrap();
    let target = "Aa";
    let input = ["Aa", "aA", "áÁ", "Aã"];

    let matches = input.into_iter().filter(|s| collator.compare(s, target).is_eq()).collect::<Vec<_>>();

    assert_eq!(&matches, &["Aa, Aã"])
}
```

Variable `matches` contains `["Aa", "aA", "áÁ", "Aã"]`, which is incorrect since the `CaseLevel::On` option should make "Aa" not equal to "aA/áÁ".

### PR DESCRIPTION
**Pull request by hsivonen**
Fixes #2884

### CODE REVIEW

File: components/collator/tests/tests.rs [hunk 1]
```diff
@@ -1404,6 +1404,20 @@ fn test_conformance_non_ignorable() {
...
+fn test_case_level() {
+    let mut options = CollatorOptions::new();
+    options.strength = Some(Strength::Primary);
+    options.case_level = Some(CaseLevel::On);
+    let collator_with_case =
+        Collator::try_new_unstable(&icu_testdata::unstable(), &Default::default(), options)
+            .unwrap();
+    assert_ne!(
+        collator_with_case.compare("aA", "Aa"),
+        core::cmp::Ordering::Equal
```
Comment (echeran) [inline components/collator/tests/tests.rs:1417]:
```suggestion
        core::cmp::Ordering::Less
```

**Review by echeran — DISMISSED** [review]

File: components/collator/tests/tests.rs (hunk 1 above)
Comment (hsivonen) [reply components/collator/tests/tests.rs:1417]: This is `assert_ne`.

Comment (hsivonen) [reply components/collator/tests/tests.rs]: Incorporated the suggestion in the fixup.

**Review by markusicu — APPROVED** [review]
lgtm tnx

**Pull request merged**